Source/WebCore/dom/Document.cpp

```diff
@@ -4065,8 +4065,9 @@ const URL& Document::urlForBindings() const
             return false;
 
         auto areSameSiteIgnoringPublicSuffix = [](StringView domain, StringView otherDomain) {
-            auto domainString = topPrivatelyControlledDomain(domain.toStringWithoutCopying());
-            auto otherDomainString = topPrivatelyControlledDomain(otherDomain.toStringWithoutCopying());
+            auto& publicSuffixStore = PublicSuffixStore::singleton();
+            auto domainString = publicSuffixStore.topPrivatelyControlledDomain(domain.toStringWithoutCopying());
```

Comment:
Not new to this patch but I think we should consider updating `topPrivatelyControlledDomain()` to take in a `StringView` instead of a `String`. Having to call `toStringWithoutCopying()` is a usual sign of this.